Chore add local stack docker support 2

Egor250625 opened the pull request:
Добавлен профиль application-local-stack для запуска приложения в локальном Docker окружении.
Создан Dockerfile для сборки образа приложения.
README обновлён с инструкцией по запуску локального стека.

Egor250625 committed:
init: initial project

Egor250625 committed:
docs: add README.md

Egor250625 committed:
chore: add Dockerfile for project image

Egor250625 committed:
docs:add Docker build and run instructions to README.md

DarkRubin commented on Dockerfile:
```diff
@@ -0,0 +1,9 @@
+FROM openjdk:21-jdk
+WORKDIR /app
+COPY --from=builder /app/build/libs/*.jar app.jar
+ENTRYPOINT ["java","-jar","app.jar"]
```
Добавь также EXPOSE 8080 для проброса порта из контейнера

DarkRubin reviewed (changes requested):
Доделай пока предыдущую задачу, потом дополнительно проверю функционал этой

Egor250625 committed:
fix: add EXPOSER 8080 to DockerFile

Egor250625 committed:
Merge branch 'dev' into chore-add-local-stack-docker-support-2

# Conflicts:
#	.gitignore
#	README.md
#	build.gradle

Egor250625 committed:
fix: change application-local-stack.yml,delete wrong files and change application-ide.yml

DarkRubin commented on src/main/resources/db/changelog/db.changelog-1.1.sql:
```diff
@@ -0,0 +1,3 @@
+--changeset egorivanov:1.1
```
Снова появилась пустая миграция

DarkRubin commented on src/main/resources/application-local-stack.yml:
```diff
@@ -0,0 +1,6 @@
+spring:
+  datasource:
+    url: jdbc:postgresql://localhost:5432/postgres
```
В local-stack укажи host url-a к БД используя алиас контейнера `posgres` вместо `localhost`

DarkRubin commented on src/main/resources/application-local-stack.yml:
```diff
@@ -0,0 +1,6 @@
+    url: jdbc:postgresql://localhost:5432/postgres
+    username: ${POSTGRES_USERNAME}
+    password: ${POSTGRES_PASSWORD}
+    driver-class-name: org.postgresql.Driver
```
Драйвер для postgresql указывать не обязательно

DarkRubin commented on Dockerfile:
```diff
@@ -0,0 +1,10 @@
+FROM gradle:8-jdk21 AS builder
+WORKDIR /app
+COPY . .
```
Также можно копировать не все содержимое репозитория, а только `build.gradle` и `settings.gradle`для загрузки зависимостей. И `src` для сборки jar файла

Egor250625 committed:
fix: change settings in DockerFile

fix: make url in application-local-stack.yml correct

fix: delete migration from project

Egor250625 committed:
fix: make correct root in build.gradle

Egor250625 committed:
fix: make correct root in main package
